fastboot-protocol/src/protocol.rs

```diff
@@ -90,10 +97,10 @@ impl<'a> FastBootResponse {
     fn from_parts(resp: &str, data: &'a str) -> Result<Self, FastBootResponseParseError> {
         trace!("Parsing Response: {} {}", resp, data);
         match resp {
-            "OKAY" => Ok(Self::Okay(data.into())),
-            "INFO" => Ok(Self::Info(data.into())),
-            "TEXT" => Ok(Self::Text(data.into())),
-            "FAIL" => Ok(Self::Fail(data.into())),
+            "OKAY" => Ok(Self::Okay(str_cut_null(data).into())),
```

Comment:
This is too late as we already parse the data into a `&str` at this point, while the garbage data could be non-utf8 in theory making from_bytes panic (something to fix, seems i missed that when cleaning up my early experiments). 

fwiw you need to handle the data in the same way as `u32::from_str_radix` will get unhappy if the str contains trailing non-digits.  Note that DATA is also an ascii hex digit, it's not raw data bytes as you allude to in your commit message.

So the right solution probably is leave this dode as is. But change from_bytes to cut the response data at the first `\0` byte before further processing